compiler/damlc/tests/src/DA/Test/DamlcIntegration.hs

```diff
@@ -132,27 +136,31 @@ getIntegrationTests registerTODO scenarioService version = do
     createDirectoryIfMissing True outdir
 
     dlintDataDir <- locateRunfiles $ mainWorkspace </> "compiler/damlc/daml-ide-core"
-    let opts = (defaultOptions (Just version))
-          { optThreads = 0
-          , optCoreLinting = True
-          , optDlintUsage = DlintEnabled dlintDataDir False
-          }
 
     -- initialise the compiler service
     vfs <- makeVFSHandle
-    damlEnv <- mkDamlEnv opts (Just scenarioService)
     -- We use a separate service for generated files so that we can test files containing internal imports.
-    pure $
+    let tree :: TestTree
+        tree = askOption $ \(LfVersionOpt version) ->
+          let opts = (defaultOptions (Just version))
+                { optThreads = 0
+                , optCoreLinting = True
+                , optDlintUsage = DlintEnabled dlintDataDir False
+                }
+          in
+          withResource (mkDamlEnv opts (Just scenarioService)) (\_damlEnv -> pure ()) $ \mkDamlEnv ->
```

Comment:
I find that name shadowing pretty confusing, maybe rename to `getDamlEnv`?

Reply:
:+1: